### PR TITLE
feat(FEC-10835): expose share to plugin

### PR DESCRIPTION
**Pull request by Yuvalke**
### Description of the Changes

Expose share component to plugin, related issues:
Issues: 
1. have multiple overlay states for CVAA and Share.
2. related components weren't exported.
3. CVAA doesn't behave correctly with the keyboard since the state wasn't taken into account keyboard components.

Solution: 
1. add a new prop to overlay to handle it inside it.
2. export all relevant components.
3. Overlay will be handle once and have a common state so it'll be fixed.

### CheckLists

- [ ] changes have been done against master branch, and PR does not conflict
- [ ] new unit / functional tests have been added (whenever applicable)
- [ ] test are passing in local environment
- [ ] Travis tests are passing (or test results are not worse than on master branch :))
- [ ] Docs have been updated

### CODE REVIEW

**Review by yairans — DISMISSED** [review]

File: src/components/copy-button/copy-button.js [hunk 1]
```diff
@@ -15,6 +17,15 @@ const TIMEOUT = 2000;
 
 const COMPONENT_NAME = 'CopyButton';
 
+const ICON_COPY_SVG = [
```
Comment (Yuvalke) [inline src/components/copy-button/copy-button.js:20]: change svg to path

File: src/components/copy-button/copy-button.js [hunk 2]
```diff
@@ -93,25 +104,28 @@ class CopyButton extends Component {
...
+              tabIndex="0"
+              ref={el => {
+                if (props.addAccessibleChild) {
+                  props.addAccessibleChild(el);
+                }
+              }}
+              className={copyUrlClasses}
+              onClick={this.copy}
+              onKeyDown={this.onKeyDown}>
+              <Icon id="copy" color="#fff" path={ICON_COPY_SVG} state={IconState.INACTIVE} width="24" height="24" viewBox="0 0 24 24" />
```
Comment (Yuvalke) [inline src/components/copy-button/copy-button.js:124]: remove inactive state, it's the defualt

File: src/components/copy-button/copy-button.js [hunk 3]
```diff
@@ -93,25 +104,28 @@ class CopyButton extends Component {
...
-          onClick={this.copy}
-          onKeyDown={this.onKeyDown}
-          title={<Text id="copy.button" />}>
-          <Icon type={IconType.Copy} />
-          <Icon type={IconType.Check} />
-        </a>
+        <ButtonControl name={COMPONENT_NAME}>
+          <Tooltip label={<Text id="copy.button" />}>
+            <Button
+              role="link"
```
Comment (Yuvalke) [inline src/components/copy-button/copy-button.js:114]: remove

File: src/styles/_buttons.scss [hunk 4]
```diff
@@ -51,3 +51,41 @@
...
+  height: 32px;
+  background-color: transparent;
+
+  &.only-icon {
+    padding: 4px;
+    width: 32px;
+    height: 32px;
+  }
+
+  &.no-icon {
```
Comment (Yuvalke) [inline src/styles/_buttons.scss:73]: only-text

Comment (Yuvalke) [reply src/styles/_buttons.scss:73]: only-label

File: src/reducers/shell.js [hunk 5]
```diff
@@ -201,6 +203,12 @@ export default (state: Object = initialState, action: Object) => {
         smartContainerOpen: action.open
       };
 
+    case types.UPDATE_OVERLAY:
```
Comment (yairans) [inline src/reducers/shell.js:206]: can we move it to overlay reducer as this is overlay action?

Comment (Yuvalke) [reply src/reducers/shell.js:206]: we can but it changes the shell prop, it seems like it better to handle all shell in one component

**Review by yairans — APPROVED** [review]

**Pull request merged**